src/handlers/rentals.ts

```diff
@@ -55,6 +56,17 @@ export function handleAssetRented(event: AssetRented): void {
     currentRental.save()
   }
 
+  let lastAssetNonceUpdateHistoryId = getIndexesUpdatesHistoryNextCount().value.minus(BigInt.fromI32(1))
```

Comment:
`getAssetIndexesUpdatesHistoryCount` is the function that you're looking for here.

Reply:
`getAssetIndexesUpdatesHistoryCount` returns the index of the next `AssetNonceUpdateHistory`. Taking into consideration that the event to increase the nonce has already been processed, the current index is the one that you would like to set. The previous indexes might not be relevant or might not exist as the index 0 is not generated through an event, but the number 1 is. If you want the previous index to be set, be aware to check individually for its existence as the current if will not set the index as `RENT` if the other one doesn't exist.
You can use `getAssetIndexesUpdatesHistoryCount` to get the current index.

Reply:
After re-reviewing the code, you were right. The `getIndexesUpdatesHistoryCount` is the function to use. The other indexes are related to the entity type that is described in their parent entity, the index history entity.